src/provider/provider_os_memory.c

```diff
@@ -191,6 +198,33 @@ static umf_result_t translate_params(umf_os_memory_provider_params_t *in_params,
         return result;
     }
 
+    result =
+        os_translate_memory_flag(in_params->visibility, &provider->visibility);
+    if (result != UMF_RESULT_SUCCESS) {
+        LOG_ERR("incorrect memory flag: %u", in_params->visibility);
+        return result;
+    }
+
+    provider->fd = os_create_anonymous_fd(provider->visibility);
+    if (provider->fd == -1) {
+        LOG_PERR(
+            "creating an anonymous file descriptor for memory mapping failed");
+        return UMF_RESULT_ERROR_UNKNOWN;
+    }
+
+    provider->size_fd = 0; // will be increased during each allocation
+    provider->max_size_fd = get_max_file_size();
+
+    errno = 0;
+    if (provider->fd > 0 &&
```

Comment:
what if provider->fd == 0? you check only -1.
Also, could you move calling os_set_file_size outside if condition and use additional var to hold the ret value? This way it would be easier to debug and analyse the code if needed.

Reply:
1) It is OK. The following code should be executed only if `provider->fd > 0`
2) Done

Reply:
@bratpiorka Done